packages/peregrine/lib/talons/CheckoutPage/useShippingMethod.js

```diff
@@ -81,28 +81,37 @@ export const useShippingMethod = props => {
         }
     }, [primaryShippingAddress]);
 
-    // Grab the selected shipping method from the primary shipping address.
+    // Grab the selected shipping method. In order:
+    // 1. From the results of our specific query to fetch it
+    // 2. The lowest cost shipping method
     const selectedShippingMethod = useMemo(() => {
+        let selectedMethod;
+
+        // From the results of our specific query to fetch it.
         try {
-            let selectedMethod =
-                primaryShippingAddress.selected_shipping_method;
-
-            if (!selectedMethod) {
-                // If there are shipping methods to choose from,
-                // pick the lowest cost one from there instead.
-                if (shippingMethods.length) {
-                    // We know that the shipping methods are sorted by price,
-                    // so the first one is the lowest cost one.
-                    selectedMethod = shippingMethods[0];
-                }
-            }
+            selectedMethod =
+                chosenShippingMethodData.cart.shipping_addresses[0]
+                    .selected_shipping_method;
+        } catch (err) {
+            // We don't have data from our specific query to fetch the selected shipping method.
+            // Intentionally swallow this error.
```

Comment:
Since we hide logs in production we can and should log the error _just in case_. Swallowing errors can make future us sad.

```js
if (process.env.NODE_ENV === 'development') {
  console.log(err);
}

Reply:
Fixed in https://github.com/magento/pwa-studio/pull/2371/commits/7318d2bd0602c7c5bbbacad19747c57247c7b0fa.

This was never really an error worth showing anyone: I was using that `try / catch` as a way not to have to check each object in `chosenShippingMethodData.cart.shipping_addresses[0].selected_shipping_method;` for truthiness but it turns out the only one we really have to check for is `chosenShippingMethodData` itself, so I just went ahead and did that.